beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1992,6 +2020,128 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         Ok(())
     }
 
+    /// Runs the `map_fn` with the committee cache at `shuffling_epoch` if the head of the chain is
+    /// `head_block_root`.
```

Comment:
There's something strange about this comment, the "if" implies it might only work for the canonical head. Maybe instead we could say something like:

```
Runs the `map_fn` with the committee cache for `shuffling_epoch` from the chain with head `head_block_root`.
```